src/main/java/mate/academy/spring/dao/impl/UserDaoImpl.java

```diff
@@ -20,11 +20,26 @@ public UserDaoImpl(SessionFactory factory) {
     public Optional<User> findByEmail(String email) {
         try (Session session = factory.openSession()) {
             Query<User> findByEmail = session.createQuery(
-                    "FROM User WHERE email = :email", User.class);
+                    "FROM User u "
+                            + "LEFT JOIN FETCH u.roles r "
+                            + "WHERE u.email = :email", User.class);
             findByEmail.setParameter("email", email);
             return findByEmail.uniqueResultOptional();
         } catch (Exception e) {
             throw new DataProcessingException("User with email " + email + " not found", e);
         }
     }
+
+    @Override
+    public Optional<User> get(Long id) {
+        try (Session session = factory.openSession()) {
+            Query<User> getByUser = session.createQuery("FROM User u "
+                    + "LEFT JOIN FETCH u.roles r "
+                    + "WHERE u.id = :id", User.class);
+            getByUser.setParameter("id", id);
+            return Optional.of(getByUser.getSingleResult());
+        } catch (Exception e) {
+            throw new DataProcessingException("Not found user for id " + id, e);
+        }
```

Comment:
Do you really need to return user roles, when you want just to get info about the user?